dev-docs/RFCs/v5.0/attribute-animation-rfc.md

```diff
@@ -9,41 +9,128 @@ Notes:
 
 ## Motivation
 
-For a general motivation of animation, see the [Property Animation RFC]().
+For a general motivation of animation, see the [Property Animation RFC](./property-animation-rfc.md).
 
-The goal of this supplmentary RFC is to investigate to what extend deck.gl could provide automatic interpolation of vertex attribute arrays, in addition to primitive values like properties, uniforms and gl parameters.
+The goal of this supplementary RFC is to investigate to what extend deck.gl could provide automatic interpolation of vertex attribute arrays, in addition to primitive values like properties, uniforms and gl parameters.
 
 ## Proposal: Attribute Array Interpolation
 
-TBA
+### Behavior
 
+When attribute updates are triggered by data change or updateTriggers, instead of immediately applying the new values, transition the current values to the new values by user-specified duration and easing curve.
 
+### APIs
 
-## Proposal: Layer accessors can be constant values
-
-> NOTE: Copied from feature scratch-pad, but related to attribute animation, would allow simple interpolation.
-
-In cases where there is a direct correspondence between an accessor and a vertex attributes  would allow for a nice optimization where a vertex attribute could simply be set to a single value (using a so called “generic vertex attribute” from luma.gl)
-
-Allowing an accessor to be set to a value instead of a function could be a nice way to enable generic vertex attributes. I.e.
+In deck.gl layer classes, certain attributes can declare themselves to be "animatable", which allows for smooth interpolation when their values are updated. This will be desirable for attributes such as sizes, widths, positions and colors. Animation will be an opt-in feature for backward compatibility:
 ```js
-   new Layer({getColor: x => [255, 0, 0]}) // would create a full array/WebGLBuffer with colors, each set to [255, 0, 0], just like today
+this.state.attributeManager.add({
+  // Not animatable
+  pickingColors: {size: 3, type: GL.UNSIGNED_BYTE, update: this.calculatePickingColors},
+  // Animatable
+  positions: {size: 3, accessor: 'getPosition', update: this.calculatePositions, animate: true},
+  colors: {size: 4, type: GL.UNSIGNED_BYTE, accessor: 'getColor', update: this.calculateColors, animate: true}
+});
 ```
-while
+
+When creating layers, animation can be enabled by supplying an `transition` prop. Animation parameters are defined per attribute by using attribute names or accessor names as keys, similar to that of `updateTriggers`:
 ```js
-   new Layer({getColor: [255, 0, 0]}) // would set a generic vertex attribute (one value shared by all verts, not allocating any array/buffers at all)
+new Layer({
+  transition: {
+    getPositions: {
+      duration: 600
+    },
+    getColors: {
+      duration: 300,
+      easing: d3.easeCubicInOut
```

Comment:
The easing just modifying the "time" parameter so it doesn't need to be visible in GLSL.